Add beta5 section (#140)
SolarArbiter/SolarArbiter.github.io: CHANGELOG.md

Detailed changes to the Solar Forecast Arbiter Core python library can be found
in the core documentation's [what's new](https://solarforecastarbiter-core.readthedocs.io/en/latest/whatsnew/index.html) series.

## [1.0beta5] - Unreleased

### Added

> 

Reviewer: - [ ] event forecasts

- [ ] normalization

- [ ] reference forecasts

> - All report plots are now created with Plotly by default. We found plotly to
>   currently contain functionality to meet the requirements of the Arbiter's
>   usecases. Plotly's more mature javascript library also makes it easier to
>   optimize reports for display in web browsers.

Reviewer: ```suggestion
- All report plots are now created with Plotly instead of Bokeh. Plotly's JSON specification and more mature javascript library makes it easier to optimize reports for display in web browsers.
```

> - Added API endpoint to recompute report at `/reports/<report_id>/recompute`.

Reviewer: ```suggestion
- Added API endpoint to recompute report at `/reports/<report_id>/recompute`. API users can access this endpoint now, and a button will be added to the dashboard in a future release.
```

- Added API endpoint to get the actions the current user is allowed to take on
  an object at `/users/actions-on/<object_id>`.

- Added API endpoints to query for latest available data or a timerange of
  available data. These endpoints are found by appending `/latest` or
  `/timerange` to the end of a values endpoint.

### Changed

- Individual forecasts or observations can be singled out by double-clicking
  the legend of a report timeseries or scatter plot.

- Report timeseries no longer draw a line over periods of missing data.

- Improved the speed of reading aggregate values from the API.

- Improved the speed of storing timeseries values at the API.

> - Dashboard will not hide buttons for actions that the userscannot take where
>   possible.

Reviewer: ```suggestion
- Dashboard will not hide buttons for actions that the users cannot take.
```

  an object at `/users/actions-on/<object_id>`.

- Added API endpoints to query for latest available data or a timerange of
  available data. These endpoints are found by appending `/latest` or
  `/timerange` to the end of a values endpoint.

- Added support for event forecasts and associated reports. A list of supported
  metrics can be found on the [metrics page](http://localhost:4000/metrics/#metrics-for-deterministic-event-forecasts).

- Metrics now automatically determine normalization based on Observation type.
  AC power observations are normalized by AC capacity; DC power by DC capacity.
  Normalized metrics set to nan for all other variables

- Users may now select a reference forecast to use in computing  
  [forecast skill](http://localhost:4000/metrics/#s).

### Changed

- Individual forecasts or observations can be singled out by double-clicking
  the legend of a report timeseries or scatter plot.

- Report timeseries no longer draw a line over periods of missing data.

- Improved the speed of reading aggregate values from the API.

- Improved the speed of storing timeseries values at the API.

> - Dashboard will not hide buttons for actions that the users cannot take.

Reviewer: @lboeman I'm working on a PR to your branch. Can you clarify this item. Seems that the first `not` should be removed.

Author: You're right, that's a typo.

Detailed changes to the Solar Forecast Arbiter Core python library can be found
in the core documentation's [what's new](https://solarforecastarbiter-core.readthedocs.io/en/latest/whatsnew/index.html) series.

> ## [1.0beta5] - 2020-04-23

Reviewer 2: ```suggestion
## [1.0beta5] - 2020-04-24
```

Database migrations have to be done carefully, so tomorrow morning is better

Reviewer: definitely

> ## [1.0beta5] - 2020-04-24

Reviewer 2: ```suggestion
## [1.0beta5] - 2020-04-27
```

### Added

- Added support for event forecasts and associated reports. A list of supported
  metrics can be found on the [metrics page](https://solarforecastarbiter.org/metrics/#metrics-for-deterministic-event-forecasts).

- Reports now support normalized metrics such as NMAE. Normalization is
  automatically determined based on Observation type. AC power observations
  are normalized by AC capacity; DC power by DC capacity. Normalized metrics

>   set to nan for all other variables.

Reviewer: ```suggestion
  are set to nan for all other variables.
```

  all sky and cloudy sky.

- All report plots are now created with Plotly instead of Bokeh. Plotly's JSON
  specification and more mature javascript library makes it easier to optimize
  reports for display in web browsers.

- Added API endpoint to recompute report at `/reports/<report_id>/recompute`.
  API users can access this endpoint now, and a button will be added to the
  dashboard in a future release.

- Added API endpoint to get the actions the current user is allowed to take on
  an object at `/users/actions-on/<object_id>`.

- Added API endpoints to query for latest available data or a timerange of
  available data. These endpoints are found by appending `/latest` or
  `/timerange` to the end of a values endpoint.

- Added DOE ARM sites to reference database.

### Changed

- Individual forecasts or observations can be singled out by double-clicking
  the legend of a report timeseries or scatter plot.

- Dashboard hides buttons for actions that a users cannot take due to
  insufficient permissions.

- Improved the speed of reading aggregate values from the API.

- Improved the speed of storing timeseries values at the API.

> 

Reviewer: ```suggestion

- Clarified data validation and data resampling/alignment sections of reports.

```

  all sky and cloudy sky.

- All report plots are now created with Plotly instead of Bokeh. Plotly's JSON
  specification and more mature javascript library makes it easier to optimize
  reports for display in web browsers.

- Added API endpoint to recompute report at `/reports/<report_id>/recompute`.
  API users can access this endpoint now, and a button will be added to the
  dashboard in a future release.

- Added API endpoint to get the actions the current user is allowed to take on
  an object at `/users/actions-on/<object_id>`.

- Added API endpoints to query for latest available data or a timerange of
  available data. These endpoints are found by appending `/latest` or
  `/timerange` to the end of a values endpoint.

- Added DOE ARM sites to reference database.

### Changed

- Individual forecasts or observations can be singled out by double-clicking
  the legend of a report timeseries or scatter plot.

- Dashboard hides buttons for actions that a users cannot take due to
  insufficient permissions. This feature has been implemented on individual
  site, forecast, probabilistic forecast, aggregate, and observation pages.

>   This feature will be extended to tabulated listing pages in a future release.

Author: @wholmgren @alorenzo175 Does this clarify things? Should we have the note about extending this to other pages or leave it off?

Reviewer: It seems to still not quite describe the situation. Doesn't it really just apply to delete?

Author: No, it extends to uploading and downloading data, adding observations to an aggregate, adding permissions to a role, and granting or revoking a role. Would it just be simpler if I created a list of these here?

Reviewer: maybe it's ok to leave the existing text if you mention the future extention also applies to the create permissions
